bson/bsonrw/extjson_wrappers.go

```diff
@@ -430,17 +430,20 @@ func (ejv *extJSONValue) parseTimestamp() (t, i uint32, err error) {
 
 		switch val.t {
 		case bsontype.Int32:
-			if val.v.(int32) < 0 {
-				return 0, fmt.Errorf("$timestamp %s number should be uint32: %s", key, string(val.v.(int32)))
+			value := val.v.(int32)
+
+			if value < 0 {
+				return 0, fmt.Errorf("$timestamp %s number should be uint32: %d", key, value)
```

Comment:
I switched the `%s` modifiers to `%d` because naively casting numbers to strings yields un-printable characters.